pkgs/top-level/aliases.nix

```
@@ -1175,6 +1175,7 @@ mapAliases ({
   partition-manager = libsForQt5.partitionmanager; # Added 2024-01-08
   pash = throw "'pash' has been removed: abandoned by upstream. Use 'powershell' instead"; # Added 2023-09-16
   patchelfStable = patchelf; # Added 2024-01-25
+  paup = lib.warn "'paup' has been renamed to 'paup-cli'" paup-cli; # Added 2024-09-11
```

This should just be a direct alias and eventually a throw after and undetermined amount of time

```suggestion
  paup = paup-cli; # Added 2024-09-11
```

---

-1. Having it initially be a warn allows for configs to continue to work for a while (the point of it being a direct alias for a while) while ensuring that when the config is run, developers will know to update the reference before their config stops working.

---

I personally don't see too much value in raising a warning, as it would be fine for users to use the alias until it gets removed at some point. We don't really have stability guarantees on unstable, and people upgrading between stable revisions are expected to read the release notes and deal with errors in their configurations.

I think it would be interesting to offer some sort of `warnAliases` option in nixpkgs, that would raise a warning every time an alias is used. But that would be out of scope for this PR

---

we need a general fix for this and not everyone trying to fix it on their own.

---

FWIW, when I brought this up in the GNOME update PR, it was met with positive feedback and applied: https://github.com/NixOS/nixpkgs/pull/333911#discussion_r1777463411